### ISSUE TITLE
c++: fatal error: Killed signal terminated program cc1plus

### ISSUE THREAD

**Issue by carbontwelve** [VROOM-Project/vroom-docker#84]
Looks like its been unable to build the docker container for the past three months https://github.com/VROOM-Project/vroom-docker/actions/workflows/master_push.yml

I tried building locally with and got similar error output to the Github Action. My local Docker install was limited to 1GB RAM and upon increasing that to 4GB the build successfully completed.

**Comment by jcoupey** [comment]
Thanks for reporting. I have no idea what is the RAM limit on the automated build and whether we have a way to try increasing it. We may hear from @nilsnolde if he has some availability. ;-)

**Comment by kkarbowiak** [comment]
Apart from possible RAM limit, the builds seems to fail due to switch to `std::format`, but this may be a separate issue.
https://github.com/VROOM-Project/vroom-docker/actions/runs/10649730597/job/29520274807#step:6:321

**Comment by jcoupey** [comment]
That's a good point. From the above link, it seems like the underlying image is still Ubuntu 22.04. In which case the gcc version there does not support C++20 format.

**Comment by rodda-kyusu** [comment]
Which version of Ubuntu supports the C++20 format?  Am running into the C++20 issue locally when building as well in docker.

**Comment by rodda-kyusu** [comment]
Tried this snippet in a github action:

```
jobs:
 build:
  runs-on: ubuntu-24.04
```

Got a different (but super related) error

```
#18 ERROR: process "/bin/sh -c echo \"Cloning and installing vroom release/branch ${VROOM_RELEASE}...\" &&     git clone --branch $VROOM_RELEASE  --single-branch --recurse-submodules https://github.com/VROOM-Project/vroom.git &&     cd vroom &&     make -C /vroom/src -j$(nproc)" did not complete successfully: exit code: 2
#16 [linux/arm64 builder 2/5] RUN echo "Updating apt-get and installing dependencies..." &&   apt-get -y update > /dev/null && apt-get -y install > /dev/null   git-core   build-essential 	g++   libssl-dev 	libasio-dev   libglpk-dev 	pkg-config
#16 CANCELED
------
 > [linux/amd64 builder 4/5] RUN echo "Cloning and installing vroom release/branch master..." &&     git clone --branch master  --single-branch --recurse-submodules https://github.com/VROOM-Project/vroom.git &&     cd vroom &&     make -C /vroom/src -j$(nproc):
6.179 g++ -MMD -MP -I. -std=c++20 -Wextra -Wpedantic -Wall -O3 -DASIO_STANDALONE -DUSE_ROUTING=true -D USE_LIBGLPK=true -c main.cpp -o main.o
6.181 g++ -MMD -MP -I. -std=c++20 -Wextra -Wpedantic -Wall -O3 -DASIO_STANDALONE -DUSE_ROUTING=true -D USE_LIBGLPK=true -c algorithms/kruskal.cpp -o algorithms/kruskal.o
6.466 In file included from main.cpp:26:
6.466 utils/version.h:13:10: fatal error: format: No such file or directory
6.466    13 | #include <format>
6.466       |          ^~~~~~~~
6.469 compilation terminated.
6.474 make: *** [makefile:76: main.o] Error 1
6.474 make: *** Waiting for unfinished jobs....
7.543 make: Leaving directory '/vroom/src'
------
 2 warnings found (use docker --debug to expand):
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 1)
 - FromAsCasing: 'as' and 'FROM' keywords' casing do not match (line 30)
Dockerfile:24
--------------------
  23 |     
  24 | >>> RUN echo "Cloning and installing vroom release/branch ${VROOM_RELEASE}..." && \
  25 | >>>     git clone --branch $VROOM_RELEASE  --single-branch --recurse-submodules https://github.com/VROOM-Project/vroom.git && \
  26 | >>>     cd vroom && \
  27 | >>>     make -C /vroom/src -j$(nproc)
  28 |     
--------------------
```

**Comment by rodda-kyusu** [comment]
Was able to get this to build in Docker using `job: ubuntu-24.04`

And the base image: `gcc:14.2-bookworm`.  Clearly suboptimal given this isn't the slim image, but was able to unblock what I was working on.

**Comment by kkarbowiak** [comment]
> Was able to get this to build in Docker using `job: ubuntu-24.04`

There is no need to change the `runs-on` entry as this is just what the GitHub Runner uses as the OS. The relevant part is the OS set in Dockerfile.

> And the base image: `gcc:14.2-bookworm`. Clearly suboptimal given this isn't the slim image, but was able to unblock what I was working on.

AFAIK, this is just the image used for building the binary, so the only cost of non-slim image is the download time.

**Comment by jcoupey** [comment]
We need to build with gcc >= 13, https://github.com/VROOM-Project/vroom-docker/pull/85#issuecomment-2341245075

**Comment by joewragg** [comment]
Hi I have a fix but I cannot create a branch on this github repo as I lack permission?

Essentially changing this line https://github.com/VROOM-Project/vroom-docker/blob/master/Dockerfile#L1
to :
`FROM ubuntu:24.04 as builder` or `FROM debian:trixie-slim as builder`

**Comment by jcoupey** [comment]
Sure, feel free to send a PR. It does make sense to update the debian version at some point anyway.

**Comment by joewragg** [comment]
> Sure, feel free to send a PR. It does make sense to update the debian version at some point anyway.

I would love to but it doesn't look like I have permission to create a branch?

**Comment by kkarbowiak** [comment]
@joewragg: Please create a fork of this repo (using one of the top buttons on repo's main page), clone the fork locally, create changes, push to your forked repo and create a PR there.

**Comment by joewragg** [comment]
> @joewragg: Please create a fork of this repo (using one of the top buttons on repo's main page), clone the fork locally, create changes, push to your forked repo and create a PR there.
@jcoupey @kkarbowiak 

PR: https://github.com/VROOM-Project/vroom-docker/pull/88

**Comment by jcoupey** [comment]
Fixed in #88, thanks @joewragg for the PR!
